hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -690,6 +690,39 @@ private SerializerAdapter lookupJavaSerializer(Class type) {
         return null;
     }
 
+    /**
+     * Makes sure that the classes registered as Compact serializable are not
+     * overriding the default serializers, if the
+     * {@link #allowOverrideDefaultSerializers} configuration option is set to
+     * {@code false}.
+     * <p>
+     * Must be called in the constructor of the child classes after they
+     * complete registering default serializers.
+     */
+    protected void verifyDefaultSerializersNotOverriddenWithCompact() {
+        // If the user explicitly set to override default serializers, we should
+        // respect that and allow it to register Compact serializers for such
+        // types. No need to perform further checks.
+        if (allowOverrideDefaultSerializers) {
+            return;
+        }
+
+
+        for (Class clazz : compactStreamSerializer.getCompactSerializableClasses()) {
+            if (!constantTypesMap.containsKey(clazz)) {
+                continue;
+            }
+
+            throw new IllegalArgumentException("Compact serializer for the "
```

Comment:
```suggestion
            throw new IllegalArgumentException("Compact serializer for the type "
```

Reply:
did something similar to that, but used the word `class` instead